Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -1980,13 +1980,110 @@ public AdvanceDeclineVolumeRatio ADVR(IEnumerable<Symbol> symbols, Resolution? r
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to casting the input value to a TradeBar</param>
+        /// <returns>The Advance/Decline Difference indicator for the requested symbol over the specified period</returns>
+        [DocumentationAttribute(Indicators)]
+        public AdvanceDeclineDifference ADD(IEnumerable<Symbol> symbols, Resolution? resolution = null, Func<IBaseData, TradeBar> selector = null)
+        {
+            var name = CreateIndicatorName(QuantConnect.Symbol.None, "A/D Difference", resolution ?? GetSubscription(symbols.First()).Resolution);
+            var add = new AdvanceDeclineDifference(name);
+            foreach (var symbol in symbols)
+            {
+                add.Add(symbol);
+                InitializeIndicator(symbol, add, resolution, selector);
+            }
+
+            return add;
+        }
+
+        /// <summary>
+        /// Creates a new McClellan Oscillator indicator
+        /// </summary>
+        /// <param name="symbols">The symbols whose McClellan Oscillator we want</param>
+        /// <param name="fastPeriod">Fast period EMA of advance decline difference</param>
+        /// <param name="slowPeriod">Slow period EMA of advance decline difference</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to casting the input value to a TradeBar</param>
+        /// <returns>The McClellan Oscillator indicator for the requested symbol over the specified period</returns>
+        [DocumentationAttribute(Indicators)]
+        public McClellanOscillator MO(IEnumerable<Symbol> symbols, int fastPeriod = 19, int slowPeriod = 39, Resolution? resolution = null, Func<IBaseData, TradeBar> selector = null)
+        {
+            return MO(symbols.ToArray(), fastPeriod, slowPeriod, resolution, selector);
+        }
+
+        /// <summary>
+        /// Creates a new McClellan Oscillator indicator
+        /// </summary>
+        /// <param name="symbols">The symbols whose McClellan Oscillator we want</param>
+        /// <param name="fastPeriod">Fast period EMA of advance decline difference</param>
+        /// <param name="slowPeriod">Slow period EMA of advance decline difference</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to casting the input value to a TradeBar</param>
+        /// <returns>The McClellan Oscillator indicator for the requested symbol over the specified period</returns>
+        [DocumentationAttribute(Indicators)]
+        public McClellanOscillator MO(Symbol[] symbols, int fastPeriod = 19, int slowPeriod = 39, Resolution? resolution = null, Func<IBaseData, TradeBar> selector = null)
```

Comment:
We should rename it `MOSC` since the other oscillators' name end with `OSC`: `ADOSC` and `ULTOSC`.